distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -281,3 +102,149 @@ type Marshalling_Error
+        Vector.from_polyglot_array proxy
+
+    ## Convert the object to a JSON
+    to_js_object : JS_Object
+    to_js_object self = self
+
+    ## Convert to string
+    to_text : Text
+    to_text self = Json.stringify self
+
+    ## Checks if this JS_Object is equal to another JS_Object.
+
+       Arguments:
+       - that: The map to compare `self` to.
+    == : JS_Object -> Boolean
+    == self that = case that of
+        _ : JS_Object ->
+            self_keys = self.field_names
+            that_keys = that.field_names
+            self_keys.length == that_keys.length && self_keys.all key->
+                (self.get key == that.at key).catch No_Such_Key.Error _->False
+        _ -> False
+
+    ## Render the object for display
+    render : Integer -> Integer -> Text
+    render self depth=0 max_depth=5 = render self depth max_depth
+
+render object depth max_depth = case object of
+    _ : JS_Object ->
+        if depth == max_depth then '"{...}"' else
+            keys = object.field_names.take 10
+            children = keys.map key-> key.to_json + ': ' + (render (object.get key) (depth+1) max_depth)
+            '{' + (children.join ", ") + '}'
+    _ : Vector ->
+        if depth == max_depth then '"[...]"' else
+            children = 0.up_to (10.min object.length) . map i-> render (object.at i) (depth+1) max_depth
+            "[" + (children.join ", ") + "]"
+    _ : Array ->
+        proxy = Array_Proxy.new object.length (i-> make_enso (object.at i))
+        @Tail_Call render (Vector.from_polyglot_array proxy) depth max_depth
+    Nothing -> object.to_json
+    _ : Text -> object.to_json
+    _ : Boolean -> object.to_json
+    _ : Number -> object.to_json
+    _ -> @Tail_Call render (JS_Object.Value object) depth max_depth
+
+make_enso js_object =
+    case js_object of
+        Nothing -> Nothing
+        _ : Text -> js_object
+        _ : Boolean -> js_object
+        _ : Number -> js_object
+        _ : Array ->
+            proxy = Array_Proxy.new js_object.length (i-> make_enso (js_object.at i))
+            Vector.from_polyglot_array proxy
+        _ : Vector ->
+            proxy = Array_Proxy.new js_object.length (i-> make_enso (js_object.at i))
+            Vector.from_polyglot_array proxy
+        _ : JS_Object -> js_object
+        _ -> JS_Object.Value js_object
+
+make_javascript enso_object =
+    if Warning.get_all enso_object != [] then make_javascript (Warning.set enso_object []) else
+        case enso_object of
+            _ : JS_Object ->
+                enso_object.field_names.fold new_object current->key->
+                    value = enso_object.get key
+                    js_value = make_javascript value
+                    set_value current key js_value
+                    current
+            _ : Array -> Vector.from_polyglot_array enso_object . map make_javascript
+            _ : Vector -> enso_object.map make_javascript
+            _ -> enso_object
```

Comment:
I'd appreciate a note that warnings are stripped from the returned value.

Also I'm not sure if the `if` is really needed? We can just always call `Warning.set enso_object []` - it will not be an error if there were no warnings in the first place, right?